### ISSUE TITLE
[pickers] Screen reader not announcing month year when changing month

### ISSUE THREAD

**Issue by m4theushw** [mui/mui-x#4475]
### Current behavior 😯

When changing the month, the screen reader doesn't announce the new month year. The label even has [`aria-live`](https://developer.mozilla.org/en-US/docs/Web/Accessibility/ARIA/ARIA_Live_Regions) but it doesn't seem to work.

<img width="500" src="https://user-images.githubusercontent.com/42154031/162970212-0a49d241-a3b4-4421-a73a-ddfd216e8033.gif" alt="nvda_was8z9eHJg" style="max-width: 100%;">

### Expected behavior 🤔

It should announce the new month year when clicking the previous or next button.

When I go to the previous or next month in https://www.w3.org/TR/wai-aria-practices/examples/dialog-modal/datepicker-dialog.html, it works.

<img width="500" src="https://user-images.githubusercontent.com/42154031/162968956-c263916c-8e12-436d-9157-6dd7ac4265de.gif" alt="nvda_SS7OS6jB1R" style="max-width: 100%;">

### Steps to reproduce 🕹

Steps:

1. Go to https://mui.com/x/react-date-pickers/date-picker/#basic-usage and use NVDA
2. Open the calendar
3. Navigate between months with the buttons

### Context 🔦

Noticed while reviewing #4472

**Comment by LukasTy** [comment]
Closing this issue as the problem seems to have been resolved.

https://github.com/mui/mui-x/assets/4941090/42c41942-c359-455a-824e-0f810b60540b
